runtime/src/bank.rs

```diff
@@ -2764,6 +2765,22 @@ impl Bank {
         }
     }
 
+    #[allow(dead_code)]
+    fn hash_rewards_into_partitions(
```

Comment:
doesn't look like this function needs to be defined on bank. maybe move it into the `epoch_rewards_hasher` module to avoid contributing to the monstrosity that is `bank.rs`?  tests can move too if so

Reply:
I think it is better to leave this function on the bank. 
It uses `StakeReward` struct, which is defined on the bank.

Reply:
> It uses StakeReward struct, which is defined inside bank.

this is a separate problem. that type has no business being specified in the bank module and should be moved (in a different PR)

Reply:
> in a different PR

@t-nelson I assume you mean in a pr that goes in first and this pr is rebased once that pr goes in?
What .rs file do you think `StakeReward` belong in?

Reply:
> > in a different PR
> 
> @t-nelson I assume you mean in a pr that goes in first and this pr is rebased once that pr goes in? What .rs file do you think `StakeReward` belong in?

new one? `runtime/src/stake_rewards.rs` maybe.

i'm indifferent on whether that change happens before or after this one

Reply:
#32014 is intended to address this

Reply:
#32014 is merged. I rebased this pr on master after that merge into master.